frontend/src/Form/beta/ModelCardForm.tsx

```diff
@@ -0,0 +1,120 @@
+import { Box, Divider, Stack, StepButton, StepLabel, Stepper } from '@mui/material'
+import MaterialStep from '@mui/material/Step'
+import { useTheme } from '@mui/material/styles'
+import Form from '@rjsf/mui'
+import { DescriptionFieldProps } from '@rjsf/utils'
+import validator from '@rjsf/validator-ajv8'
+import React, { Dispatch, SetStateAction, useState } from 'react'
+import CustomTextInput from 'src/MuiForms/CustomTextInput'
+import TagSelector from 'src/MuiForms/TagSelectorBeta'
+import { setStepState } from 'utils/beta/formUtils'
+
+import { SplitSchemaNoRender } from '../../../types/interfaces'
+import Nothing from '../../MuiForms/Nothing'
+
+export default function ModelCardForm({
+  splitSchema,
+  setSplitSchema,
+  canEdit = false,
+}: {
+  splitSchema: SplitSchemaNoRender
+  setSplitSchema: Dispatch<SetStateAction<SplitSchemaNoRender>>
+  canEdit?: boolean
+}) {
+  const [activeStep, setActiveStep] = useState(0)
+  // TODO - add validation
+  const [_openValidateError, _setOpenValidateError] = useState(false)
+
+  const theme = useTheme()
+
+  const currentStep = splitSchema.steps[activeStep]
+
+  if (!currentStep) {
+    return null
+  }
+
+  const onFormChange = (form) => {
+    if (form.schema.title !== currentStep.schema.title) {
+      return
+    }
+
+    setStepState(splitSchema, setSplitSchema, currentStep, { ...currentStep.state, ...form.formData })
+  }
+
+  function DescriptionFieldTemplate(_props: DescriptionFieldProps) {
+    return <></>
+  }
+
+  return (
+    <Stack
+      direction={{ xs: 'column', sm: 'row' }}
+      spacing={4}
+      justifyContent='left'
+      divider={<Divider flexItem orientation='vertical' />}
+      sx={{ width: '100%' }}
+    >
+      <Box>
+        <Stepper
+          id='form-page-stepper'
+          activeStep={activeStep}
+          nonLinear
+          alternativeLabel
+          orientation='vertical'
+          connector={<Nothing />}
+          sx={{ minWidth: 'max-content' }}
+        >
+          {splitSchema.steps.map((step, index) => (
+            <MaterialStep key={step.schema.title}>
+              <StepButton sx={{ p: 0, m: 0 }} onClick={() => setActiveStep(index)} icon={<Nothing />}>
+                <StepLabel
+                  sx={{
+                    padding: 0,
+                    '& .MuiStepLabel-label': {
+                      fontSize: '16px !important',
+                    },
+                    '& .Mui-active': {
+                      borderBottomStyle: 'solid',
+                      color: `${theme.palette.primary.main} !important`,
```

Comment:
Should also avoid using `!important` here and the line below too

Reply:
This is needed otherwise the style won't update

Reply:
That's because the color is likely defined elsewhere higher up and this is not the right place to set it.

Either that or it's defined lower down with `!important` and if this is the case then we have no choice.